srml/contracts/src/gas.rs

```diff
@@ -205,7 +204,7 @@ pub fn buy_gas<T: Trait>(
 	let gas_available = T::BlockGasLimit::get() - <Module<T>>::gas_spent();
 	if gas_limit > gas_available {
 		// gas limit reached, revert the transaction and retry again in the future
-		return Err(BLOCK_FULL);
+		return Err("gas block limit reached");
```

Comment:
This `BLOCK_FULL` acted as a special value to ultimately signal the block production logic that the block is full. As far as I understand, this was used because there is no other way to convey that signal.

Looking at the code, this message no longer has any semantics. Is there a way to restore previous behavior here? Is it worth it given that we would want to extend and tap onto the weight mechanism?

cc @tomusdrw

Reply:
indeed, this needs to be `TemporarilyExhausted` or something similar.

Reply:
Yes!

Reply:
I added a `CheckBlockGasLimit` signed extension. That needs to be added to the transaction and makes sure that the gas limit is not hit.